internal/impl/pure/cache_bloom.go

```diff
@@ -0,0 +1,498 @@
+		Stable().
+		Summary(`Stores keys in a bloom in-memory filter, useful for deduplication. This cache is therefore reset every time the service restarts.`).
+		Description(`This provides the bloom package which implements a fixed-size thread safe filter.
+
+A Bloom filter is a concise/compressed representation of a set, where the main requirement is to make membership queries; i.e., whether an item is a member of a set. A Bloom filter will always correctly report the presence of an element in the set when the element is indeed present. A Bloom filter can use much less storage than the original set, but it allows for some 'false positives': it may sometimes report that an element is in the set whereas it is not.
+
+When you construct, you need to know how many elements you have (the desired capacity), and what is the desired false positive rate you are willing to tolerate. A common false-positive rate is 1%. The lower the false-positive rate, the more memory you are going to require. Similarly, the higher the capacity, the more memory you will use.
+
+It uses the package ` + "[`bloomfilter`](github.com/bits-and-blooms/bloom/v3)" + `
+
+The field ` + bloomCacheFieldInitValuesLabel + ` can be used to pre-populate the memory cache with any number of keys:
+
+` + "```yml" + `
+cache_resources:
+  - label: foocache
+    bloom:
+      cap: 1024
+      init_values:
+        - foo
+        - bar
+` + "```" + `
+
+These values can be overridden during execution.`).
+		Field(service.NewIntField(bloomCacheFieldCapLabel).
+			Description("The cache maximum capacity (number of entries)").
+			Default(bloomCacheFieldCapDefaultValue)).
+		Field(service.NewFloatField(bloomCacheFieldFalsePositiveRateLabel).
+			Description("false positive rate. 1% is 0.01").
+			Default(bloomCacheFieldFalsePositiveRateDefaultValue)).
+		Field(service.NewStringListField(bloomCacheFieldInitValuesLabel).
+			Description("A table of keys that should be present in the cache on initialization. This can be used to create static lookup tables.").
+			Default([]string{}).
+			Example([]string{
+				"Nickelback",
+				"Spice Girls",
+				"The Human League",
+			})).
+		Field(service.NewObjectField(bloomCacheFieldStorageLabel,
+			service.NewStringField(commonFieldStoragePathLabel).
+				Description(`Path to a dir or file where we can restore or write dumps of bloom filter.
+
+This cache can try to dump the content of the cache in disk during the benthos shutdown.
+
+Also, the cache can try to restore the state from an existing dump file. Errors will be ignored on this phase.
+
+This field accepts two kinds of value:
+
+If the path contains a single file with extension '.dat', it will be used for I/O operations.
+
+If the path constains a directory, we will try to use the most recent dump file (if any).
+
+If necessary, we will create a file with format 'benthos-bloom-dump.<timestamp>.dat'
+`).
+				Examples("/path/to/bloom-dumps-dir", "/path/to/bloom-dumps-dir/dump.dat").
```

Comment:
Is `/path/to/bloom-dumps-dir/dump.dat` a valid example given the description?
> If necessary, we will create a file with format 'benthos-bloom-dump.<timestamp>.dat'

Reply:
Hello, I will expand this section to be more clear.
In fact we can specify any file or directory but if we do not specify a file, we will generate one using the given convention + timestamp nanosecond to avoid conflicts.

my plan is to prepare flushes over time, to be possible, for instance, keep filters of 24 hours and use a sliding window of 7 days. It means if we can dedupe events in the last seven days.

also I’d like to add a periodic cleansing to wipe the cache should be nice

however… using files to manage state has some challenges. We may need one directory per benthos instance to avoid clashing, etc. I just want to add something really simple and if it doesn’t not fit, people can send a pull request to enhance it.

Using redis is “easier” because we can set a ttl on each bloom or cuckoo index.

I have a specific problem to solve that is dedup batches of entries without use a lot of memory.  However each time the benthos start it will find an empty filter…

we will see, besides dedupe I can’t see any other usage of bloom / cuckoo